Standards/scs-0112-v1-sonic.md

```diff
@@ -0,0 +1,99 @@
+---
+title: SONiC Support in SCS
+type: Decision Record # Standard  | Decision Record | Procedural
+status: Draft
+track: Global # Global | IaaS | Ops | KaaS | IAM
+description: |
+  SCSS-0112 outlines architecutral decisions in SCS in regards to SONi support and integration.
+---
+
+<!---
+This is a template striving to provide a starting point for
+creating a standard or decision record adhering to scs-0001.
+Replace at least all text which is _italic_.
+See https://github.com/SovereignCloudStack/standards/blob/main/Standards/scs-0001-v1-sovereign-cloud-standards.md
+--->
+
+## Introduction
+
+SONiC support in [SCS](https://scs.community) was considered within the context of [VP04 Networking](https://scs.community/tenders/lot4), sub-lot 1 SDN scalability. Different challenges and approaches to SDN scalability have been explored and more specifically those who require support in the underlay network. Using SONiC in the underlay can have benefits for SCS users by using a standardized OS for network devices and also having a clear path for network scalability when using SONiC. For this to work, we have to define how SONiC is used and supported architecturally in SCS. This document outlines the architectural decisions in regards to SONiC support and integration in SCS.
+
+## Motivation
+
+In respect to SDN scalability improvements in Openstack and SCS, there are several ways SONiC can be leveraged.
+
+### SONiC as a network OS where dynamic network configuration in Openstack is required
+
+In many network designs for Openstack, configuration of the actual network hardware by Openstack Neutron service is required. The following network designs apply:
+
+- VLANs. Uisng VLANs to segment tenant networks requires the network switch to be configured. This can be manual or dynamic configuration via the ML2 Neutron driver.
+
+- EVPN/VXLAN on the switch. In this use case, SONiC runs on leaf switches. Leafs terminal VXLAN endpoints and run BGP/EVPN for the control plane. Again, the ML2 Neutron driver is used to dynamically configure the network switch. The link between the switch and the service is regular VLAN.
+
+- VXLAN on servers and switches. This is a hybrid use case, where most of the SDN to pushed to the server, but the network is still involved where support for bare metal hosts is needed.
+
+### Automation and tooling for SONiC
+
+There is no tooling in SCS or Openstack communities to facilitate with the rollout and configuration of enterprise scale SONiC deployments. Netbox and OSISM can be integrated, so that Netbox becomes the source of truth for network configuration and OSISM supports the initial rollout and configuration for the switches.
```

Comment:
```suggestion
There is no tooling in SCS or Openstack communities to facilitate the rollout and configuration of enterprise scale SONiC deployments. Netbox and OSISM can be integrated, so that Netbox becomes the source of truth for network configuration and OSISM supports the initial rollout and configuration for the switches.
```